go/staking/api/sanity_check.go

```diff
@@ -228,6 +229,22 @@ func (g *Genesis) SanityCheck(now epochtime.EpochTime) error { // nolint: gocycl
 		return fmt.Errorf("staking: sanity check failed: %w", err)
 	}
 
+	tokenSymbolLength := len(g.TokenSymbol)
+	if tokenSymbolLength == 0 {
+		return fmt.Errorf("staking: sanity check failed: token symbol is empty")
+	}
+	if tokenSymbolLength > TokenSymbolMaxLength {
+		return fmt.Errorf("staking: sanity check failed: token symbol exceeds maximum length")
+	}
+	match := regexp.MustCompile("^[A-Z]+$").FindString(g.TokenSymbol)
```

Comment:
This regexp should probably be an exported variable at the api level (together with the max length/max value.

Reply:
Agreed. Done.